lib/templ.ml

```diff
@@ -118,20 +118,73 @@ let setup_link conf =
 let esc s = (Util.escape_html s :> string)
 
 let rec eval_variable conf =
+  let urlv_aux evar str =
+  (* rebuild the current url from conf.env, replacing &evar=xxx by &evar=str () *)
+    let url = (Util.commd conf :> string) in
+    let url = (List.hd (String.split_on_char '?' url)) ^ "?" in
```

Comment:
please do not use List.hd and use a match instead:
```
let url = 
    match String.split_on_char '?' url with
    | [] -> error??
    | s::_l -> s ^ "?"

Reply:
thanks!

actually I just realized that it was okay to use `List.hd` here since the doc says "`String.split_on_char` can not gives an empty list".

but I think it's still better practice to always a match, that way we are sure everything is good without having to look to much into it.

also you don't need the pipe and the sprintf here ;)